fuzzers/test_fuzzers.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for fuzzers/."""
+import os
+import importlib
+
+import pytest
+
+from common import utils
+
+# pylint: disable=invalid-name,unused-argument
+
+
+def get_all_fuzzer_dirs():
+    """Returns the list of all fuzzers."""
+    fuzzers_dir = os.path.join(utils.ROOT_DIR, 'fuzzers')
+    return [
+        fuzzer for fuzzer in os.listdir(fuzzers_dir)
+        if (os.path.isfile(os.path.join(fuzzers_dir, fuzzer, 'fuzzer.py')) and
+            fuzzer != 'coverage')
+    ]
+
+
+def _get_fuzzer_module(fuzzer):
+    """Get the module for |fuzzer|'s fuzzer.py."""
+    return 'fuzzers.{}.fuzzer'.format(fuzzer)
+
+
+def _get_all_fuzzer_modules():
+    """Returns the fuzzer.py modules for all fuzzers."""
+    return [
+        importlib.import_module(_get_fuzzer_module(fuzzer))
+        for fuzzer in get_all_fuzzer_dirs()
+    ]
+
+
+@pytest.mark.parametrize('fuzzer_module', _get_all_fuzzer_modules())
+def test_build_function_errors(fuzzer_module, fs):
+    """Test that calling the build function will cause an error when
+    the benchmark can't be built. Test also ensures that there are no
+    type errors which can be made by having the wrong signature."""
+    with pytest.raises(Exception) as error:
+        fuzzer_module.build()
```

Comment:
I dont understand these tests. Does this mean it should always error out when calling the build. Shouldn't we have the test the other way that it always passes when you call the fuzz and build functions.

Reply:
>I dont understand these tests.

The tests intentionally call build/fuzz under circumstances in which they should fail and then asserts that they do in fact fail.
The reason I added this test is because when I was adding the run feature to CI, I found that most fuzzers as they were written did not exception when the command they executed returned nonzero. 

>Does this mean it should always error out when calling the build. 

Exactly.

>Shouldn't we have the test the other way that it always passes when you call the fuzz and build functions.

Ideally yes, but it isn't practical to test that everything works, it's easier to test that they fail properly.
We have this in CI, not locally though because it would take too long to build and run every fuzzer.

Reply:
The `build` test is probably unnecessary but it probably doesn't hurt to have (except hurting test time since we are running each fuzzer as a different test (e.g. one for HF, one for AFL etc.)).

Reply:
What does the test failure look like, is it just timing out ? Should we test that exception is timeout one, as otherwise catching any test can be some more important problem.

Reply:
No it isn't just timing out.
I don't know if we should specify which exception is thrown because I don't think it's important to enforce this. I just want to ensure that when building is impossible an exception is thrown.
In most cases it will probably be subprocess.CalledProcessError when utils.build_benchmark is called, but again i don't consider it important that this happens. I only consider it important that they throw exceptions of some kind.